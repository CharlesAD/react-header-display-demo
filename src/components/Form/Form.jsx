import PropTypes from "prop-types";

export default function Form({ setName }) {
  const handleSubmit = (event) => {
    event.preventDefault();

    const submission = Object.fromEntries(new FormData(event.target));
    setName(submission.name);

    event.target.reset();
  };

  return (
    <form className="mx-auto w-48" onSubmit={handleSubmit}>
      <div>
        <label htmlFor="name" className="sr-only">
          Name
        </label>
        <input
          type="text"
          placeholder="Enter your name"
          id="name"
          className="m-5 rounded border"
          name="name"
        />
      </div>

      <button
        type="submit"
        className="flex justify-center bg-purple-500 px-4 py-2 text-white"
      >
        Submit
      </button>
    </form>
  );
}

Form.propTypes = {
  setName: PropTypes.func.isRequired,
};
