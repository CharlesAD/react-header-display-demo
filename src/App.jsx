import Profile from "components/Profile/Profile";
import { useState } from "react";
import "./App.css";
import Products from "./components/Products/Products";
import Form from "./components/Form/Form";

const imgUrl = "https://i.imgur.com/yXOvdOSs.jpg";

const products = [
  {
    category: "Sporting Goods",
    price: "$49.99",
    stocked: true,
    name: "Football",
  },
  {
    category: "Sporting Goods",
    price: "$9.99",
    stocked: true,
    name: "Baseball",
  },
  {
    category: "Sporting Goods",
    price: "$29.99",
    stocked: false,
    name: "Basketball",
  },
  {
    category: "Electronics",
    price: "$99.99",
    stocked: true,
    name: "iPod Touch",
  },
  {
    category: "Electronics",
    price: "$399.99",
    stocked: false,
    name: "iPhone 5",
  },
  { category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7" },
];

function App() {
  const [name, setName] = useState({ name: "" });

  return (
    <>
      <h1 className=" ">Hello {name}</h1>;
      {name && <Profile name={name} imgUrl={imgUrl} />}
      <Form setName={setName} />
      <Products products={products} />
    </>
  );
}

export default App;
