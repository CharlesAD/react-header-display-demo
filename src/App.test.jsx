import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";
import Form from "./components/Form/Form";

/**
 * TODO:
 * 1. Write a snapshot test and visually inspect it to insure that it is rendering correctly
 * 2. Get by label text and submit the form.
 * 3. Expect the name that you typed in the form `toBeInTheDocument()`.
 */

it("renders without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("Get by label text and submits the form", () => {
  render(<App />);
  const rendered = renderer.create(<App />).toJSON();

  const input = screen.getByLabelText("submit");
  userEvent.type(input, "Charles");
});
