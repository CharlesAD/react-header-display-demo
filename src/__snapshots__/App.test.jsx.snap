// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <h1
    className=" "
  >
    Hello 
  </h1>,
  ";",
  <form
    className="mx-auto w-48"
    onSubmit={[Function]}
  >
    <div>
      <label
        className="sr-only"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="m-5 rounded border"
        id="name"
        name="name"
        placeholder="Enter your name"
        type="text"
      />
    </div>
    <button
      className="flex justify-center bg-purple-500 px-4 py-2 text-white"
      type="submit"
    >
      Submit
    </button>
  </form>,
  <ul>
    <li>
      Football
       - 
      $49.99
    </li>
    <li>
      Baseball
       - 
      $9.99
    </li>
    <li>
      Basketball
       - 
      $29.99
    </li>
    <li>
      iPod Touch
       - 
      $99.99
    </li>
    <li>
      iPhone 5
       - 
      $399.99
    </li>
    <li>
      Nexus 7
       - 
      $199.99
    </li>
  </ul>,
]
`;
